src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java

```diff
@@ -63,68 +65,48 @@ protected SignMock(@NotNull Block block)
 	protected SignMock(@NotNull SignMock state)
 	{
 		super(state);
-
-		for (int i = 0; i < 4; i++)
-		{
-			this.lines[i] = state.getLine(i);
-		}
-		this.color = state.getColor();
-		this.glowing = state.isGlowingText();
+		this.front = new SignSideImpl(state.front);
+		this.back = new SignSideImpl(state.back);
 	}
 
 	@Override
 	public @NotNull List<Component> lines()
 	{
-		List<Component> components = new ArrayList<>();
-
-		for (String line : this.lines)
-		{
-			components.add(LegacyComponentSerializer.legacySection().deserialize(line));
-		}
-
-		return components;
+		return front.lines();
 	}
 
 	@Override
 	public @NotNull Component line(int index) throws IndexOutOfBoundsException
 	{
-		return LegacyComponentSerializer.legacySection().deserialize(getLine(index));
+		return front.line(index);
 	}
 
 	@Override
 	public void line(int index, @NotNull Component line) throws IndexOutOfBoundsException
 	{
-		Preconditions.checkNotNull(line, "Line cannot be null!");
-		this.lines[index] = LegacyComponentSerializer.legacySection().serialize(line);
+		front.line(index, line);
```

Comment:
Why was the Precondition removed?

Reply:
The method is calling `SignSideMock#line(int, Component)` which contains the null check. So it would be redundant keeping it there.